site/content/docs/howto/configure-auth-for-webapps.md

```diff
@@ -0,0 +1,347 @@
+---
+title: Using the Pinniped Supervisor to provide authentication for web applications
+description: Allow your Kubernetes cluster users to authenticate into web apps using the same identities.
+cascade:
+  layout: docs
+menu:
+  docs:
+    name: Web Application Authentication
+    weight: 800
+    parent: howtos
+---
+The Pinniped Supervisor is an [OpenID Connect (OIDC)](https://openid.net/connect/) issuer that can be used to bring
+your user identities from an external identity provider into your Kubernetes clusters for all your `kubectl` users.
+It can also be used to bring those same identities to web applications that are intended for use by the same users.
+For example, a Kubernetes dashboard web application for cluster developers could use the Supervisor as its OIDC
+identity provider.
+
+This guide explains how to use the Supervisor to provide authentication services for a web application.
+
+## Prerequisites
+
+This guide assumes that you have installed and configured the Pinniped Supervisor, and configured it with an
+external identity provider, as described in the other guides.
+
+This guide also assumes that you have a web application which supports configuring an OIDC provider for user
+authentication, or that you are developing such a web application. From the point of view of the Supervisor,
+your webapp is called a "client" ([as defined in the OAuth 2.0 spec](https://www.rfc-editor.org/rfc/rfc6749#section-1.1)).
+
+Typically, the web application should use the OIDC client support from its web application development
+framework (e.g. Spring, Rails, Django, etc.) to implement authentication. The Supervisor requires that:
+- Clients must use the [OIDC authorization code flow](https://openid.net/specs/openid-connect-core-1_0.html#CodeFlowAuth).
+  Clients must
+  use `code` as the [response_type](https://openid.net/specs/openid-connect-core-1_0.html#AuthorizationExamples)
+  at the authorization endpoint.
+- Clients must use [PKCE](https://oauth.net/2/pkce/) during the authorization code flow.
+- Clients must be confidential clients, meaning that they have a client ID and client secret.
+  Clients must use [client secret basic auth](https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1)
+  for authentication at the token endpoint.
+- Clients must use `query` as the
+  [response_mode](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest) at the authorization endpoint,
+  or not specify the `response_mode` param, which defaults to `query`.
+
+Most web application frameworks offer all these capabilities in their OAuth2/OIDC libraries.
+
+## Create an OIDCClient
```

Comment:
Maybe this section should also have a new subsection which discusses the performance implications of using OIDCClients (all the bcrypts) and how the admin may like to adjust the Supervisor pod's cpu requests and limits once they are familiar with usage patterns of their Supervisor.